x-pack/filebeat/input/httpjson/internal/v2/config_request.go

```diff
@@ -88,6 +88,7 @@ type requestConfig struct {
 	RedirectMaxRedirects   int               `config:"redirect.max_redirects"`
 	RateLimit              *rateLimitConfig  `config:"rate_limit"`
 	Transforms             transformsConfig  `config:"transforms"`
+	ProxyURL               string            `config:"proxy_url"`
```

Comment:
```suggestion
	ProxyURL               *urlConfig            `config:"proxy_url"`
```

If you change to the `urlConfig` type, if will handle the validation and parsing of the url automatically.